app/assets/src/components/views/bulk_download/ChooseStep.jsx

```diff
@@ -1,16 +1,55 @@
 import React from "react";
 import PropTypes from "~/components/utils/propTypes";
-import { find, filter, get, some, map, isUndefined } from "lodash/fp";
+import {
+  debounce,
+  find,
+  filter,
+  get,
+  some,
+  map,
+  isUndefined,
+  orderBy,
+} from "lodash/fp";
 import cx from "classnames";
+import memoize from "memoize-one";
 
 import Dropdown from "~ui/controls/dropdowns/Dropdown";
 import LoadingMessage from "~/components/common/LoadingMessage";
 import RadioButton from "~ui/controls/RadioButton";
+import { getBackgrounds, getTaxonWithReadsSuggestions } from "~/api";
 import PrimaryButton from "~/components/ui/controls/buttons/PrimaryButton";
 
 import cs from "./choose_step.scss";
 
+const AUTOCOMPLETE_DEBOUNCE_DELAY = 200;
+
 class ChooseStep extends React.Component {
+  state = {
+    backgroundOptions: null,
+    taxonWithReadsOptions: null,
+    loadingTaxonWithReadsOptions: false,
```

Comment:
similar for `loadingTaxonWithReadsOptions`.. if it's called `Options` I'm not usually expecting a bool